### PR TITLE
Added methods str() and cat()

### LINKED ISSUE #1: New Method: cat(arg1, arg2,..., argN)
This will be the most frequently used method, it allows the addition of one or many strings to the buffer. 
```js
var sb = new StringBuilder();
sb.cat(‘hello’);
sb.cat(‘ Javascript’, ‘ crazy’, ‘ world’).cat( ‘!!!’);
```
Note that you can send one or more arguments to the cat method and that you can instantly call the cat method again, thanks to the cascade or chaining pattern that you wisely implemented.
- Function as parameters
The parameters on the cat method shall accept function objects, this function objects references will be called by the cat method and its return value will be added to the buffer instead of the function itself. Don’t forget to bind the function to the StringBuilder, that way, the StringBuilder methods will be accessibles inside the function.
```js
var sb = new StringBuilder();
sb.cat(‘this is the first’’,’ line’, ‘\n’)
    .cat(‘here is the second’)
    .cat(‘and then’, ‘the third’)
    .cat(‘now’, function(){ return ‘ we can make some calcs’; }, ‘ here’); 
```
Note: If you are clever enough, you will use the cat method as the only one to access the buffer storage, that way you will avoid repeating yourself and immediately will add support for functions as parameters to the other methods of the API (rep, catIf, etc).

- Array as parameter
At this time the cat method supports a list of arguments which are expected to be strings, functions (which will be called inside the cat method and the return value will be pushed to the buffer) and any other kind of object that will be evaluated using the toString() method.
Having said that, your API will have to support the following call:
```js
var sb = new StringBuilder();
sb
  .cat(‘this’, [‘ is’, ‘ a’, ‘ string’], ‘ in the’)
  .cat(‘ array’, ‘ :-)’);
```
But, you would not have gotten to this point if you haven't already implemented the functions as parameters requirement. :-)
```js
var sb = new StringBuilder();
sb
  .cat(‘this’, [‘ is’, function(){ return [‘ a’, ‘ function that’,
  ‘ returns an array’]; }], ‘ ;)’);
```

### LINKED ISSUE #4: New Method: string()
This method is, in the end, the major purpose of why we are writing this library.  It will return a concatenated string of all the parameters that we have given our object.
```js
var sb = new StringBuilder();
sb.cat(‘hello’,’ world, ‘!’);
console.log(sb.string());
```
In this case, the method shall return the ‘hello world!’ string.

### PR DESCRIPTION
**Pull request by islas27**
- Added both methods unit tests
- Added examples for cat()
- Added documentation
- Added a dependency

Closes #1 & closes #4

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module.exports = stringHelper
+require('array-extensions')
+
+function stringHelper () {
+  let buffer = (arguments.length > 0) ? [...arguments].flatten() : []
```
Comment (ryanez) [inline index.js]: What if you use your own `cat` function here?

Comment (islas27) [reply index.js]: Improved this bit on 6643b30

File: examples/cat.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+let StringHelper = require('../index')
+
+let helper = new StringHelper()
+let result = helper.cat('this is the first line', '\n')
+    .cat('here is the second\n')
+    .cat(['and then', ' the third'], '\n')
+    .cat('now').cat(function () {
```
Comment (ryanez) [inline examples/cat.js]: You can use some arrows here `.cat('now').cat(() => ' we can make some calcs ')`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+module.exports = stringHelper
+require('array-extensions')
+
+function stringHelper () {
+  let buffer = []
+  let that = this
+
+  function classifier (input) {
+    if (Array.isArray(input)) return 'array'
```
Comment (ryanez) [inline index.js]: You can always use the ternary operator
```js
function classifier(input) {
  return Array.isArray(input) ? 'array' : typeof input
}
```

Or if you like the fanciness 
```js
const classifier = input => Array.isArray(input) ? 'array' : typeof input
```

File: index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+module.exports = stringHelper
+require('array-extensions')
+
+function stringHelper () {
+  let buffer = []
+  let that = this
+
+  function classifier (input) {
```
Comment (ryanez) [inline index.js]: You can implement this `function classifier { ... }` outside of `stringHelper`

File: index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    if (Array.isArray(input)) return 'array'
+    else return typeof input
+  }
+
+  this.cat = function () {
+    const len = arguments.length
+    for (let i = 0; i < len; i += 1) {
+      let arg = arguments[i]
+      let type = classifier(arg)
+      if (type === 'function') this.cat(arg.apply(that))
```
Comment (ryanez) [inline index.js]: Is it `this` or `that`? choose a side!

File: index.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      else buffer.push(arg)
+    }
+    return this
+  }
+
+  this.str = function () {
+    return buffer.join('')
+  }
+
+  this.cat([...arguments])
```
Comment (ryanez) [inline index.js:26]: the `.cat` is actually returning `this` it isn't?

File: index.js (hunk 5 above)
Comment (islas27) [reply index.js]: This! Fixed in 5ea3ee1

File: index.js (hunk 3 above)
Comment (islas27) [reply index.js]: I like the fancy way, added in 5ea3ee1

File: index.js (hunk 4 above)
Comment (islas27) [reply index.js]: indeed, fixed in 5ea3ee1

File: examples/cat.js (hunk 2 above)
Comment (islas27) [reply examples/cat.js]: Alright, used a shorter syntax, but added a note in the documentation about the usage of arrow functions and access to the helper. 5ea3ee1

File: index.js (hunk 6 above)
Comment (islas27) [reply index.js:26]: Indeed it is, removed the return statement in 5ea3ee1

File: README.md [hunk 7]
```diff
@@ -55,4 +55,6 @@ This function will 'save' all the strings sent to it in any format, be it a list
 
 **Example**: [cat.js](examples/cat.js)
 
-**Notes**: When `typeof` of an input evaluates to `object` or `undefined`, this argument will be ignored. This happens to any kind of object, map, null or undefined values. To know more about the `typeof` operand, follow this link: [typeof - MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
+**Notes**:
+- When `typeof` of an input evaluates to `object` or `undefined`, this argument will be ignored. This happens to any kind of object, map, null or undefined values. To know more about the `typeof` operand, follow this link: [typeof - MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
+- While arrow functions maybe more clean and concise, the do not bind the `this` object, so in case you need to use the helper inside a function that is going to be passed as an argument, write an old fashion function. More info on arrow functions: [Arrow Functions - MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
```
Comment (ryanez) [inline README.md]: the do not bind the `this` ? did you want to say "they"?

Comment (islas27) [reply README.md]: yeah, a typo... and tried to fix a little the way I redacted the explanation. What do you think @ryanez? Commit 7a89758

**Comment by ryanez** [comment]
LGTM 👍

**Pull request merged**